phi/eval/eval.py

```diff
@@ -123,16 +131,14 @@ def get_accuracy_evaluator(self) -> Agent:
             response_model=AccuracyResult,
         )
 
-    def run(self, answer: Optional[Union[str, Callable]] = None) -> Optional[EvalResult]:
+    def run(self, question: str, expected_answer: str, answer: Optional[Union[str, Callable]] = None) -> Optional[EvalResult]:
         logger.debug(f"*********** Evaluation Start: {self.eval_id} ***********")
 
         answer_to_evaluate: Optional[RunResponse] = None
         if answer is None:
```

Comment:
Rather switch the if-else around to test the positive case first. This is a bit of an anti-pattern

Reply:
i.e. do `if answer:`